src/robotFactory.js

```diff
@@ -1,10 +1,66 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
 
-class DeliveryDrone {}
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+  getInfo() {
+    return (
+      `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`
+    );
+  }
+}
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
If parameters number is greater than 2, you can split it into rows, 
it`s better to read.
Check all your code for this
```suggestion
  constructor(
    name, 
    weight, 
    coords, 
    chipVersion 
  ) {
```